### PR TITLE
[WIP] restructure+skeleton for architecture component overview

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@bmcelvee when you have some time let's work through this...  this is a partial implementation of the vision that @derekwaynecarr and I came up with in:

https://docs.google.com/document/d/13osh6-sR0cqbTxfwaJLZbpWwryO8MyEnyaddgplh7HE/edit

and

https://docs.google.com/document/d/1bScOZFnQvMUBxuyxTA5bRH0Du182kk1bHa9cRQIDsVs/edit

Notable issues:

1) all the links for the pulled over content are likely broken and new links should be added (e.g. link from the custom builds arch summary to the new custom builds tutorial, I marked some of these with TBD)
2) the 3 new arch files i added are showing up in the table of contents, I don't want them to (I just want to link to them).  Let's talk about what the options are for this.
3) Need to vet the components(name/grouping) in the list and their ordering
4) need blurbs for every component bullet in the main component list (we need to get these from the owning teams)
5) Need detailed pages for most/all of the component bullets
6) I still don't love the Architecture hierarchy, i'd like to discuss how we can improve it (I have some ideas)

/cc @derekwaynecarr

File: _topic_map.yml [hunk 1]
```diff
@@ -44,6 +44,12 @@ Topics:
   File: architecture
 - Name: Available cluster customizations
   File: customizations
+- Name: Builds
+  File: architecture-component-builds
+- Name: Image Registry
+  File: architecture-component-imageregistry
+- Name: Image streams
+  File: architecture-component-imagestreams
```
Comment (bparees) [inline _topic_map.yml]: I had to do this to get my new files to render at all, but it results in them being put directly into the Architecture hierarchy which i don't want.

**Comment by bparees** [comment]
also i added my 3 new files as direct content rather than modules out of expediency, i'm guessing you'll want to move them into modules and create new files that include the modules.  That just seemed like a lot of hassle :)

File: architecture/architecture.adoc [hunk 2]
```diff
@@ -17,13 +17,15 @@ becomes available.
...
 include::modules/installation-options.adoc[leveloffset=+2]
 
-include::modules/update-service-overview.adoc[leveloffset=+2]
+include::modules/update-service-overview.adoc[leveloffset=+3]
 
-include::modules/node-management.adoc[leveloffset=+2]
+include::modules/node-management.adoc[leveloffset=+3]
 
-include::modules/node-types.adoc[leveloffset=+2]
+include::modules/node-types.adoc[leveloffset=+3]
```
Comment (bparees) [inline architecture/architecture.adoc]: I attempted to align these offsets w/ the heading level of the content they contain.  Not sure if that is appropriate or not.

File: architecture/architecture-component-builds.adoc [hunk 3]
```diff
@@ -0,0 +1,137 @@
+// Module included in the following assemblies:
```
Comment (bmcelvee) [inline architecture/architecture-component-builds.adoc]: It looks like we can update existing modules and include them here:

include::modules/builds-about.adoc[leveloffset=+1]
include::modules/builds-strategy-docker-build.adoc[leveloffset=+1]
include::modules/builds-strategy-s2i-build.adoc[leveloffset=+1]
include::modules/builds-strategy-custom-build.adoc[leveloffset=+1]
include::modules/builds-strategy-pipeline-build.adoc[leveloffset=+1]

We also have a module for BuildConfigs (that may need to be updated as well): 
include::modules/builds-buildconfig.adoc[leveloffset=+1]

File: architecture/architecture-component-imageregistry.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * n/a
+
+[id='architecture-component-imageregistry']
```
Comment (bmcelvee) [inline architecture/architecture-component-imageregistry.adoc]: It looks like we can update the existing module(s) and include them here:

include::modules/registry-integrated-openshift-registry-registry-options.adoc[leveloffset=+1]

Do you want to mention any of the other registries (third party, Quay, registry.redhat.io)?

Comment (bparees) [reply architecture/architecture-component-imageregistry.adoc]: not under product architecture no.

File: architecture/architecture-component-imagestreams.adoc [hunk 5]
```diff
@@ -0,0 +1,169 @@
+// Module included in the following assemblies:
```
Comment (bmcelvee) [inline architecture/architecture-component-imagestreams.adoc]: It looks like we can update the existing module(s) and include them here:

include::modules/images-imagestream-use-image-streams-managing.adoc[leveloffset=+1]

(There are more modules for this section. I'll add them later.)

**Comment by bmcelvee** [comment]
This may not be the most up-to-date version, but if not, I'll rebuild: http://file.rdu.redhat.com/~bmcelvee/13032019/architecture/

**Comment by bparees** [comment]
@bmcelvee ok i think this is as far as i can take this right now.  I suggest we find a way to land as much of this as possible as quickly as possible so teams can start adding their pieces to it.

What's not great about it currently:

1) files are not named or modularized well.
2) numerous TBD placeholders that need to be hidden or completed.

What is good about it:

Fixes the architecture section to have a better outline and place for teams to provide more details.
Establishes a pattern for how teams should do that by introducing detailed architecture sections for builds, image streams, and image registry.

**Comment by bmcelvee** [comment]
@bparees - I brought your changes forward and made some additional updates in https://github.com/openshift/openshift-docs/pull/14167. Do you mind if I close this PR?

Thanks!

**Comment by bparees** [comment]
> Do you mind if I close this PR?

nope.

**Pull request closed without merging**